gulpfile.js/build.js

```diff
@@ -392,10 +463,404 @@ function buildPages(done) {
+      'landscape_ship_1280x853.jpg',
+      10
+    ),
+  ];
+
+  const configObj = {
+    time: new Date().toLocaleTimeString(),
+    timestamp: Number(new Date()),
+    // send a random list of blog items to make it also work on the cache
+    blogItems: blogItems.filter(() =>
+      Math.floor(Math.random() * Math.floor(2))
+    ),
+  };
+
+  return gulp
+    .src(`${project.paths.DIST}/examples/sources/**/*.html`)
+    .pipe(
+      through.obj(async (file, enc, callback) => {
+        const srcHTML = file.contents.toString();
+
+        env.renderString(srcHTML, configObj, (err, result) => {
+          if (err) {
+            logger.error(`Error rendering ${file.path}`);
+            return callback(err);
+          }
+
+          file.contents = Buffer.from(result);
+          callback(null, file);
+        });
+      })
+    )
+    .pipe(gulp.dest((f) => f.base))
+    .on('end', () => {
+      done();
+    });
+}
+
+/**
+ * Compiles the pages into standalone static files
+ *
+ * @return {Promise}
+ */
+async function staticify(done) {
+  const logger = require('@lib/utils/log')('Static File Generator');
+
+  const requestPathRegex = new RegExp(
+    `${project.paths.PAGES_DEST}|(index)?.html`,
+    'g'
+  );
+
+  const generatedFormats = SUPPORTED_FORMATS.map((format) => {
+    const f = (cb) => {
+      const env = nunjucksEnv();
+
+      const f = filter([`${project.paths.PAGES_DEST}/**/*html`, 'index'], {
+        restore: true,
+      });
+
+      return gulp
+        .src(`${project.paths.PAGES_DEST}/**/*.html`)
+        .pipe(f)
+        .pipe(
+          // Render a static version of all of our pages
+
+          through.obj(async (file, enc, callback) => {
+            const configObj = {
+              requestPath: `${file.path.replace(requestPathRegex, '')}/`,
+              format,
+              requestedFormat: format,
+            };
+
+            const srcHTML = file.contents.toString();
+
+            env.renderString(srcHTML, configObj, (err, result) => {
+              if (err) {
+                logger.error(`Error rendering ${file.path}`);
+                return callback(err);
+              }
+
+              file.contents = Buffer.from(result);
+              callback(null, file);
+            });
+          })
+        )
+        .pipe(
+          through.obj(async function (file, enc, callback) {
+            // Rewrite links inside of each of the pages
+
+            const $ = Cheerio.load(file.contents.toString());
+
+            $('a.nav-link, a.ap-m-format-toggle-link').each(function () {
+              // eslint-disable-next-line no-invalid-this
+              const $this = $(this);
+              const origURL = $this.attr('href');
+
+              $this.attr('href', getUpdatedURL(origURL, format));
+            });
+
+            const renderedPage = $.root().html();
+
+            if (
+              format === FORMAT_WEBSITES &&
+              !file.path.endsWith('tools.html')
```

Comment:
add a comment to explain why not tools

Reply:
ah, [this](https://github.com/ampproject/amp.dev/blob/b1e69255e480c62ea20015342360fd36305d4226/gulpfile.js/build.js#L503) got moved around at some point. Thanks!